x-pack/plugins/spaces/public/management/spaces_grid/spaces_grid_page.tsx

```diff
@@ -326,37 +349,57 @@ export class SpacesGridPage extends Component<Props, State> {
         }),
         actions: [
           {
-            render: (record: Space) => (
+            isPrimary: true,
+            available: (rowRecord) => this.state.activeSpace?.name !== rowRecord.name,
+            render: (rowRecord: Space) => {
+              return (
+                <EuiButtonIcon
+                  href={getSpaceNavigationURL({ serverBasePath, spaceId: rowRecord.id })}
```

Comment:
I imagined that the "switch" link would change the Space to the selected one without changing the navigation, but I think the way you have it works well also.